### PR TITLE
feat: add dialog style property

### PR DESCRIPTION
**Pull request by deser**

### CODE REVIEW

**Comment by taion** [comment]
Don't do that. Just merge in `this.props.style` on https://github.com/react-bootstrap/react-bootstrap/blob/c0e02fed8d0138fcf55f09cfbbb1dbb91c8e4f84/src/Modal.js#L170.

**Comment by deser** [comment]
To my mind style prop should be passed to root component which is BaseModal. Therefore I introduced new property (like it done for CSS class - dialogClassName) .  However  don't see the full picture. If you insist on merging I'll do that

**Comment by taion** [comment]
This is not styling the dialog.

**Comment by jquense** [comment]
yeah look at the markup of the bootstrap dialog. the actual visible dialog is not the BaseModal, that is a larger more abstract div that you don't want to style.

**Comment by deser** [comment]
Done

**Comment by taion** [comment]
I think what @jquense is saying is that you don't actually want to style the outermost modal component; instead you do want to style the dialog.

**Comment by deser** [comment]
But what have been done matches my needs. I think if you are OK, guys, we can merge

**Comment by jquense** [comment]
LGTM? I can't keep track of which component is which here anymore

**Comment by taion** [comment]
We should be using the ES7 object spread syntax here.

Also, I thought you said it should be e.g. `dialogStyle` like on https://github.com/react-bootstrap/react-bootstrap/blob/master/src/ModalDialog.js#L37?

**Comment by deser** [comment]
>>I can't keep track of which component is which here anymore
What do you mean?

>>We should be using the ES7 object spread syntax here.
Where? 

>>Also, I thought you said it should be e.g. dialogStyle
Could you explain what you mean?

File: src/Modal.js [hunk 1]
```diff
@@ -150,10 +151,15 @@ const Modal = React.createClass({
       , children
       , dialogClassName
       , animation
+      , style
       , ...props } = this.props;
 
     let { modalStyles } = this.state;
 
+    if (style) {
+      modalStyles = merge(modalStyles, style);
```
Comment (taion) [inline src/Modal.js]:
```js
modalStyles = { ...modalStyles, ...style }
```

**Pull request merged**